src/bearhugStorage.factory.js

```diff
@@ -0,0 +1,59 @@
+/* global angular */
+angular
+  .module('talis.bearhug')
+  .factory('bearhugStorage', bearhugStorage);
+
+/**
+* In-memory store of state around token.
+*
+* Note that only the underlying token is stored; bearer token is derived from that.
+**/
+function bearhugStorage(bearerUtils) {
+  // encapsulated state
+  var token = null;
```

Comment:
I have not included any form of persistence.  Although such things are available, I didn't want to couple them together.  Perhaps it would make sense to provide some kind of event-publish for these changes so that a persistence layer could hook in.